### PR TITLE
fix: x-forwarded-proto is sanitized now

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Fixes an issue where someone can inject a non-standard value in the `x-forwarded-proto` header.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -30,6 +32,8 @@ export function createIPXHandler ({
   }
   const handler: Handler = async (event, _context) => {
     const host = event.headers.host
+    event.headers['x-forwarded-proto'] = X_FORWARDED_PROTO_VALUES.includes(event.headers['x-forwarded-proto'].toLowerCase()) ? event.headers['x-forwarded-proto'] : 'http'
```
Comment (nickytonline) [inline src/index.ts:35]: Now we only allow the allowed values for `x-forwarded-proto`.

**Review by dustincrogers — DISMISSED** [review]
lgtm

**Review by dustincrogers — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function createIPXHandler ({
   }
   const handler: Handler = async (event, _context) => {
     const host = event.headers.host
-    const protocol = event.headers['x-forwarded-proto'] || 'http'
+    const protocol = event.headers['x-nf-connection-proto'] || 'http'
```
Comment (nickytonline) [inline src/index.ts:33]: @mraerino mentioned this as a better option as it can't be tampered with.

File: src/index.ts [hunk 3]
```diff
@@ -30,6 +32,8 @@ export function createIPXHandler ({
   }
   const handler: Handler = async (event, _context) => {
     const host = event.headers.host
+    event.headers['x-forwarded-proto'] = X_FORWARDED_PROTO_VALUES.includes(event.headers['x-forwarded-proto'].toLowerCase()) ? event.headers['x-forwarded-proto'] : 'http'
+
     const protocol = event.headers['x-forwarded-proto'] || 'http'
```
Comment (mraerino) [inline src/index.ts:37]: seems like a complex way of handling this, wdyt about this?
```suggestion
    const protocol = event.headers['x-forwarded-proto'] === "https" ? "https" : "http"
```

Comment (nickytonline) [reply src/index.ts:37]: Yeah, that makes sense. I think my brain was laser focused on sanitizing the header. Simplicity ftw.

File: src/index.ts [hunk 4]
```diff
@@ -15,6 +15,8 @@ export interface IPXHandlerOptions extends Partial<IPXOptions> {
   responseHeaders?: Record<string, string>
 }
 
+const X_FORWARDED_PROTO_VALUES = ['http', 'https']
+
```
Comment (mraerino) [inline src/index.ts:19]:
```suggestion
```

**Review by mraerino — APPROVED** [review]

**Review by mraerino — APPROVED** [review]

**Review by dustincrogers — APPROVED** [review]

**Pull request merged**